tests/improved-sanity-test/main.yml

```diff
@@ -210,7 +242,15 @@
       tags:
         - atomic_host_check
 
-    # we remove any subscriptions after the upgrade to verify that
+    # Setup facts again. (Need to use the 'always_run' option to ensure the
+    # role runs again)
+    - role: osname_set_fact
+      tags:
+        - osname_set_fact
+        - cloud_image
+      always_run: yes
```

Comment:
Good find!  We need always_run in the other tests instead of multiple tags named slightly differently.